### PR TITLE
Constraints for maps and records

### PR DESCRIPTION
**Pull request by skogsbaer**

### CODE REVIEW

**Comment by skogsbaer** [comment]
@albsch This PR brings support for maps and records. The tests for maps are not active yet, because tally/subtyping is still buggy.

File: src/ety_main.erl [hunk 1]
```diff
@@ -48,7 +48,7 @@ parse_args(Args) ->
          {only, $o, "only", string,
             "Only typecheck these functions (given as name/arity or just the name)"},
          {log_level,  $l,   "level",    string,
-            "Minimal log level (trace,debug,info,note,warn)"}
+            "Minimal log level (trace2,trace,debug,info,note,warn)"}
```
Comment (albsch) [inline src/ety_main.erl:51]: Why the trace2 log level addition?

**Review by albsch — APPROVED** [review]

File: src/ety_main.erl (hunk 1 above)
Comment (skogsbaer) [reply src/ety_main.erl:51]: because some trace log messages totally messed up the log. They now use trace2

**Comment by skogsbaer** [comment]
@albsch I had to rebase the commits. The PR is currently reject because erlang's stdlib make use of overriding record field types. I will add this feature.

**Pull request merged**